sympy/plotting/intervalmath/interval_arithmetic.py

```diff
@@ -34,6 +34,79 @@
 from __future__ import print_function, division
 
 from sympy.simplify.simplify import nsimplify
+from sympy.core.logic import fuzzy_and, fuzzy_or, fuzzy_not, fuzzy_xor
+
+
+class intervalMembership(object):
+    """Represents a boolean expression returned by the comparison of
+    the interval object.
+
+    Parameters
+    ==========
+
+    (a, b) : (bool, bool)
+        The first value determines the comparison as follows:
+        - True: If the comparison is True throughout the intervals.
+        - False: If the comparison is False throughout the intervals.
+        - None: If the comparison is True for some part of the intervals.
+
+        The second value is determined as follows:
+        - True: If both the intervals in comparison are valid.
+        - False: If at least one of the intervals is False, else
+        - None
+    """
+    def __init__(self, a, b):
+        self._wrapped = (a, b)
+
+    def __getitem__(self, i):
```

Comment:
Add a check for one-off error if the index are beyond range

Reply:
What needs to be checked for? It this is off by one then the IndexError below should handle it.

Reply:
I had the same thought, but I think that the drawback of making python index error to propagate upward can be the error message not being descriptive.
And I have seen this [thread](https://stackoverflow.com/questions/2522005/cost-of-exception-handlers-in-python) that the cost of using exception catching block for places where it is supposed to be programmed to have no errors is cheap.